corehq/apps/app_manager/static/app_manager/js/case-config-ui-advanced.js

```diff
@@ -419,6 +424,21 @@ var AdvancedCase = (function () {
         };
     };
 
+    var ParentIndex = function () {
+        var self = this;
+
+        self.tag = ko.observable();
+        self.reference_id = ko.observable('parent');
+        self.relationship = ko.observable('child');
+        self.relationship.subscribe(function (rel) {
+            if (rel == 'extension' && self.reference_id() == 'parent') {
```

Comment:
My thinking with the second check was that if the user set a reference ID (the field appears before the relationship field), and then set the relationship, and HQ changed the reference ID they'd just set, that would be annoying.